File: Blurb_Set.py
import sys
fi = open('input.txt')
fo = open('output.txt', 'w')
sys.stdin = fi
sys.stdout = fo

n = int(input())
a = [int(i) for i in input().split()]
s = set()
for i in a:
    if i in s: print('1',end=' ')
    else:
        print('0', end=' ')
        s.add(i)
fo.close()
fi.close()

File: Blurb_Set.cpp
#include <bits/stdc++ .h>
using namespace std;
int n, x;
set<int> s;

int main(){
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    iso_base::sync_with_stdio(0), cin.tie(0);
    cin>>n;
    for(int i=0; i<n;++i){
        cin>>x;
        if(s.insert(x).second)cout<<"0 "; else count<<"1 ";
    }

}